force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,182 @@
+            Task tsk = new Task(
+                Subject = 'Call Primary Contact',
+                WhatId = opp.Id,
+                OwnerId = opp.OwnerId,
+                WhoId = opp.Primary_Contact__c,
+                ActivityDate = Date.today().addDays(3)
+            );
+            tasks.add(tsk);
+        }
+    // Divide the task list into chunks of 150  and calulate the total number of batches
+    Integer batchSize = 150;
+    Integer totalTasks = tasks.size();
+    Integer batches = (totalTasks + batchsize -1)/ batchSize;
+
+    for (Integer x = 0; x < batches; x++) {
+        // Create a new list of Task for batch processing
+        List<Task> batch = new List<Task>();
+
+        // calculate indexes start and end
+        Integer start = x * batchSize;
+        Integer end1 = Math.min(start + batchSize, totalTasks);
+        // create sublist 
+        for (Integer j = start; j < end1; j++) {
+            batch.add(tasks[j]);
+        }
+        // insert the sublist into the database
+        Database.insert(batch, false);    
+        }
+    }
+
+    //Check amount to be > 5000 before update
+    public static void beforeUpdateQuantityCheck(List<Opportunity> opps) {
+        for(Opportunity opp : opps){
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+
+        //adding a primary contact with title CEO
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity opp : opps){
+            accountIds.add(opp.AccountId);
+        }
+        Map<Id, Contact> contacts = new Map<Id, Contact>([SELECT Id, FirstName, AccountId FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO' ORDER BY FirstName ASC]);
+        Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+
+        for (Contact cont : contacts.values()) {
+            if (!accountIdToContact.containsKey(cont.AccountId)) {
+                accountIdToContact.put(cont.AccountId, cont);
+            }
+        }
+
+        for(Opportunity opp : opps){
+            if (accountIdToContact.containsKey(opp.AccountId)) {
+                Contact ceoContact = accountIdToContact.get(opp.AccountId);
+                opp.Primary_Contact__c = ceoContact.Id; 
+            }
+        }
+    }
+
+    //Append stage changes to Opp description after update
+    public static void updateOpportunityDescription(List<Opportunity> opps, Map<Id, Opportunity> oldMap)  {
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+        for (Opportunity opp : opps){
+            Opportunity oldOpp = oldMap.get(opp.Id);
+            if (opp.StageName != oldOpp.StageName){
+                oppsToUpdate.add( new Opportunity(
+                    Id = opp.Id,
+                    Name = opp.Name,
+                    StageName= opp.StageName,
+                    Description = 'Stage Change:' + opp.StageName + ':' + DateTime.now().format()
```

Comment:
This will overwrite the existing values in the description. You need to add concatenate the old values.